connectors/sources/sharepoint.py

```diff
@@ -283,27 +303,40 @@ async def _api_call(self, url_name, url="", **url_kwargs):
                     else:
                         yield await result.json()
                     break
-            except Exception as exception:
-                if isinstance(
-                    exception,
-                    ClientResponseError,
-                ) and "token has expired" in exception.headers.get(  # pyright: ignore
+            except ClientResponseError as exception:
+                retry, retry_seconds = self._get_retry_after(
+                    retry=retry, exception=exception
+                )
+                if getattr(exception, "status", None) == 429:
+                    logger.warning("Rate limit exceeded.")
+                    if "Retry-After" not in exception.headers:  # pyright: ignore
+                        logger.warning(
+                            "Exception status is 429 but it doesn't contain Retry-After header"
+                        )
+                    retry_seconds = int(
+                        exception.headers.get("Retry-After", 30)  # pyright: ignore
+                    )
```

Comment:
I suggest changing defaulting logic so that you can specify that 429 received, Retry-After is not found therefore we will retry in 30 seconds instead. Also 30 seconds should become a constant.